R/AzureDataLake.R

```diff
@@ -0,0 +1,368 @@
+                                  azureActiveContext = azureActiveContext,
+                                  verbose = verbose)
+  if (status_code(resHttp) == 404) {
+    warning("Azure data lake response: resource not found")
+    return(NULL)
+  }
+  stopWithAzureError(resHttp)
+
+  resJsonStr <- content(resHttp, "text", encoding = "UTF-8")
+  resJsonObj <- jsonlite::fromJSON(resJsonStr)
+  resDf <- as.data.frame(resJsonObj)
+  resDf$boolean
+}
+
+#' Azure Data Lake CREATE for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
+#' @param overwrite Overwrite existing files (default FALSE).
+#' @param permission Provide the permission to be set for the directory.
+#' @param contents Provide contents to write to `relativePath`
+#' @param verbose Print tracing information (default FALSE).
+#'
+#' @return NULL
+#'
+#' @template
+#' @references
+#' @family Azure Data Lake Store functions
+#' @export
+azureDataLakeCreate <- function(azureActiveContext, azureDataLakeAccount, relativePath, overwrite = FALSE, permission = NULL, contents = "", verbose = FALSE) {
+  
+  if (!missing(azureActiveContext) && !is.null(azureActiveContext)) {
+    assert_that(is.azureActiveContext(azureActiveContext))
+    azureCheckToken(azureActiveContext)
+  }
+  assert_that(is_storage_account(azureDataLakeAccount))
+  if (!missing(permission) && !is.null(permission)) assert_that(is_permission(permission))
+  # TODO: Need a check for contents ?
+  #assert_that(is_content(contents))
+  verbosity <- set_verbosity(verbose)
+  
+  URL <- paste0(
+    "https://", azureDataLakeAccount, ".azuredatalakestore.net/webhdfs/v1/",
+    relativePath,
+    "?op=CREATE&write=true",
+    "&api-version=2016-11-01"
+  )
+  if (!missing(overwrite)  && !is.null(overwrite)) URL <- paste0(URL, "&overwrite=", overwrite)
+  if (!missing(permission)  && !is.null(permission)) URL <- paste0(URL, "&permission=", permission)
+  
+  resHttp <- callAzureDataLakeApi(URL, verb = "PUT",
+                                  azureActiveContext = azureActiveContext,
+                                  content = contents, contenttype = "text/plain; charset=UTF-8",
+                                  verbose = verbose)
+  stopWithAzureError(resHttp)
+  # TODO: Check why this returns NULL
+}
+
+#' Azure Data Lake APPEND for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
```

Comment:
File?

Reply:
Fixed in https://github.com/omkarksa/AzureSMR/pull/1 will raise a pull request once this is merged into master